OnlinerTask/OnlinerTask.BLL/Repository/MsSQLRepository.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OnlinerTask.DAL.SearchModels;
+using OnlinerTask.Data.DBModels;
+using System.Data.Entity;
+
+namespace OnlinerTask.BLL.Repository
+{
+    public class MsSQLRepository : IRepository
+    {
+        public async Task<List<ProductModel>> CheckProducts(List<ProductModel> products, string UserName)
+        {
+            await AsyncOperations.ForEachAsync(products, async i =>
+            {
+                if (await CheckItem(i.Id, UserName)) i.IsChecked = true;
+            });
+            return products;
+        }
+
+        public async Task<bool> CheckItem(int ItemId, string Username)
+        {
+            using(var context = new OnlinerProducts())
+            {
+                return await context.Products.FirstOrDefaultAsync(x=> x.ProductId == ItemId && x.UserEmail == Username) != null ? true : false;
+            }
+        }
+
+        public bool CreateOnlinerProduct(ProductModel model, string UserEmail)
+        {
+            if (model == null) return false;
+            else
+            {
+                int maxid = CreatePriceAmmount(new PriceAmmount() { Amount = model.Prices.PriceMax.Amount, Currency = model.Prices.PriceMax.Currency });
+                int minid = CreatePriceAmmount(new PriceAmmount() { Amount = model.Prices.PriceMin.Amount, Currency = model.Prices.PriceMin.Currency });
+                var product = ModelToDB(model, UserEmail, maxid, minid);
+                CreateProduct(product);
+                return true;
+            }
+        }
+
+        public int CreatePriceAmmount(PriceAmmount price)
+        {
+            if (price == null) return -1;
+            else
+            {
+                using (var context = new OnlinerProducts())
+                {
+                    context.PriceAmmounts.Add(price);
+                    context.SaveChanges();
+                    return price.Id;
+                }
+            }
+        }
+
+        public bool CreateProduct(Product product)
+        {
+            if (product == null) return false;
+            else
+            {
+                using (var context = new OnlinerProducts())
+                {
+                    context.Products.Add(product);
+                    context.SaveChanges();
+                    return true;
+                }
+            }
+        }
+
+        public bool RemoveOnlinerProduct(int itemId, string name)
+        {
+            using (var context = new OnlinerProducts())
+            {
+                var model = context.Products.Where(x => x.UserEmail == name && x.ProductId == itemId).FirstOrDefault();
+                if (model != null)
```

Comment:
It is useful when condition is swapped : if(model == null){return false;}. In this case target logic has lower nesting level.